docs/source/python_spec.rst

```diff
@@ -136,6 +153,44 @@ refer to `github.com/dmlc/dlpack <https://github.com/dmlc/dlpack>`_.
    whether the ``.device`` attribute of the array returned from ``from_dlpack`` is
    guaranteed to be in a certain order or not.
 
+.. _future-abi-compat:
+
+Future ABI Compatibility
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+ABI version 1 did not provide any fields in the structs ``DLTensor`` or
+``DLManagedTensor`` to export version info. Two equivalent structs,
+``DLTensorVersioned`` and ``DLManagedTensorVersioned``, have been added
+since ABI version 2 (DLPack version 0.7.0) and have a ``version`` field
+that can be used to export version info and check if the producer's
+DLPack version is compatible with the consumer's DLPack version. This
+section gives a path for Python libraries to upgrade to the new ABI
+(while preserving support for the old ABI):
+
+* ``__dlpack__`` should accept a ``version`` (int) keyword which is set to
+  ``None`` by default. Consumers can use this kwarg to request certain DLPack
+  versions. If ``version=None`` or ``version=60`` is requested:
+
+  * a capsule named ``"dltensor"`` which uses the old ABI (``DLTensor`` and
+    ``DLManagedTensor``) should be returned (if the producer wants to keep
+    supporting it) or
+  * a ``BufferError`` should be raised (if the producer doesn't want to keep
+    support for the old ABI)
+
+  Otherwise, a capsule named ``"dltensor"`` which uses the new ABI
+  (``DLTensorVersioned`` and ``DLManagedTensorVersioned``) should be returned.
+  If the requested version is not supported, ``__dlpack__`` should raise a
+  ``BufferError``.
+* Producers should implement a ``__dlpack_info__`` method that returns the
+  maximum supported DLPack version.
```

Comment:
```suggestion
* Producers should implement a ``__dlpack_info__`` method that returns the
  maximum supported DLPack version. If this method does not exist, the consumer
  must use the old ABI.
```

Reply:
I added this information in the point below this one (that point is all about what the consumer should do). Does that look good or do you think we should add the point here?